### ISSUE TITLE
newline-after-var rule is broken with babel-eslint >= 9

### ISSUE THREAD

**Issue by brokenmass** [eslint/eslint#11839]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10
* **npm Version:** 6.10

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es6: true,
    node: true
  },
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      globalReturn: false,
      impliedStrict: true,
      jsx: true
    },
    ecmaVersion: 2017,
    sourceType: 'module'
  },
  plugins: [
    'babel'
  ],
  "rules": {
    "newline-after-var": [
      "error",
      "always"
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const const1 = 1;
let variable1 = 1;
let variable2 = 1;
let variable3 = 1;
```

```bash
eslint --ignore-path .gitignore '**/*.js'
```

**What did you expect to happen?**
expected eslint to not return any error

**What actually happened? Please include the actual, raw output from ESLint.**
```
eslint-newline-after-var-bug/index.js
  1:1  error  Expected blank line after variable declarations  newline-after-var
  2:1  error  Expected blank line after variable declarations  newline-after-var
  3:1  error  Expected blank line after variable declarations  newline-after-var
```

**Explanation**
Since version `9.0.0`, `babel-eslint` broke the eslint rule `newline-after-var`. the reason is that since that version they have been using `@babel/parser@^7` and that changed how the code AST is generated. In the specific `let` token does not have `Keyword` type anymore but instead has type `Identifier` (this is probably because of [this](https://www.ecma-international.org/ecma-262/9.0/#sec-let-and-const-declarations-static-semantics-early-errors) so this problem is also loosely correlated to #11830)

This change affect the `newline-after-var` rule as it use the following check to group together chunks of variable declaration ([link to actual code](https://github.com/eslint/eslint/blob/master/lib/rules/newline-after-var.js#L199)):
```js
if (nextToken.type === "Keyword" && isVar(nextToken.value)) {
  return;
}
```

while the second part of the if is still valid the first now fails for `let` variables (this bug does not affects `const` or `var`)

**Related babel-eslint issue**
https://github.com/babel/babel-eslint/issues/771

**Are you willing to submit a pull request to fix this bug?**
yes if we can agree on a solution and where to implement it (`eslint`@`newline-after-var` rule or `babel-eslint`@`babylon-to-espree` code)

**Comment by g-plane** [comment]
Hey @brokenmass , thanks for the issue.

This rule has been deprecated.

**Comment by brokenmass** [comment]
@g-plane thanks for the quick response. i totally missed the fact that the rule had been deprecated.
